main.go

```diff
@@ -1,78 +1,178 @@
 	})
-
 	port := app.String(cli.StringOpt{
 		Name:   "port",
 		Value:  "8080",
 		Desc:   "Port to listen on",
 		EnvVar: "APP_PORT",
 	})
 
-	log.SetLevel(log.InfoLevel)
-	log.Infof("[Startup] methode-article-image-set-mapper is starting ")
-
-	app.Action = func() {
-		log.Infof("System code: %s, App Name: %s, Port: %s", *appSystemCode, *appName, *port)
-
-		go func() {
-			serveAdminEndpoints(*appSystemCode, *appName, *port)
-		}()
+	addresses := app.Strings(cli.StringsOpt{
+		Name:   "queue-addresses",
+		Desc:   "Addresses to connect to the queue (hostnames).",
+		EnvVar: "Q_ADDR",
+	})
 
-		// todo: insert app code here
+	group := app.String(cli.StringOpt{
+		Name:   "group",
+		Desc:   "Group used to read the messages from the queue.",
+		EnvVar: "Q_GROUP",
+	})
 
-		waitForSignal()
-	}
-	err := app.Run(os.Args)
-	if err != nil {
-		log.Errorf("App could not start, error=[%s]\n", err)
-		return
-	}
-}
+	readTopic := app.String(cli.StringOpt{
+		Name:   "read-topic",
+		Desc:   "The topic to read the meassages from.",
+		EnvVar: "Q_READ_TOPIC",
+	})
 
-func serveAdminEndpoints(appSystemCode string, appName string, port string) {
-	healthService := newHealthService(&healthConfig{appSystemCode: appSystemCode, appName: appName, port: port})
+	readQueue := app.String(cli.StringOpt{
+		Name:   "read-queue",
+		Desc:   "The queue to read the meassages from.",
+		EnvVar: "Q_READ_QUEUE",
+	})
 
-	serveMux := http.NewServeMux()
+	writeTopic := app.String(cli.StringOpt{
+		Name:   "write-topic",
+		Desc:   "The topic to write the meassages to.",
+		EnvVar: "Q_WRITE_TOPIC",
+	})
 
-	hc := health.HealthCheck{SystemCode: appSystemCode, Name: appName, Description: "Maps inline image-sets from bodies of Methode articles.", Checks: healthService.checks}
-	serveMux.HandleFunc(healthPath, health.Handler(hc))
-	serveMux.HandleFunc(status.GTGPath, status.NewGoodToGoHandler(healthService.gtgCheck))
-	serveMux.HandleFunc(status.BuildInfoPath, status.BuildInfoHandler)
+	writeQueue := app.String(cli.StringOpt{
+		Name:   "write-queue",
+		Desc:   "The queue to write the meassages to.",
+		EnvVar: "Q_WRITE_QUEUE",
+	})
 
-	if err := http.ListenAndServe(":"+port, serveMux); err != nil {
-		log.Fatalf("Unable to start: %v", err)
+	authorization := app.String(cli.StringOpt{
+		Name:   "authorization",
+		Desc:   "Authorization key to access the queue.",
+		EnvVar: "Q_AUTHORIZATION",
+	})
+	return args{
+		appSystemCode: *appSystemCode,
+		appName:       *appName,
+		port:          *port,
+		addresses:     *addresses,
+		group:         *group,
+		readTopic:     *readTopic,
+		readQueue:     *readQueue,
+		writeTopic:    *writeTopic,
+		writeQueue:    *writeQueue,
+		authorization: *authorization,
 	}
 }
 
-func waitForSignal() {
-	ch := make(chan os.Signal)
+func (m app) waitForSignals() {
+	ch := make(chan os.Signal, 1)
 	signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
 	<-ch
 }
+
+func (m app) teardown() {
+	m.queue.stop()
+	m.consumerTeardown.Wait()
```

Comment:
does this actually do anything? I don't see anywhere calls to `consumerTeardown.add(...)`